src/__test__/components/data-exploration/differential-expression-tool/DiffExprCompute.test.jsx

```diff
@@ -148,36 +148,19 @@ describe('DiffExprCompute', () => {
 
     // Choose cell set 1
     const selectCellSet1 = screen.getByRole('combobox', { name: /Compare cell set/i });
-    await act(async () => {
-      fireEvent.change(selectCellSet1, { target: { value: 'Cluster 0' } });
-    });
 
-    const cellSet1Option = screen.getByText(/Cluster 0/);
-    await act(async () => {
-      fireEvent.click(cellSet1Option);
-    });
+    userEvent.click(selectCellSet1);
+    userEvent.click(screen.getByText('Cluster 0'));
 
     // Select the 2nd cell set
     const selectCellSet2 = screen.getByRole('combobox', { name: /and cell set/i });
-    await act(async () => {
-      fireEvent.change(selectCellSet2, { target: { value: 'All' } });
-    });
-
-    const cellSet2Option = screen.getByText(/All other cells/);
-    await act(async () => {
-      fireEvent.click(cellSet2Option);
-    });
+    userEvent.click(selectCellSet2);
+    userEvent.click(screen.getByText('All other cells'));
 
     // With all samples
     const selectSampleOrGroup = screen.getByRole('combobox', { name: /within sample/i });
-    await act(async () => {
-      fireEvent.change(selectSampleOrGroup, { target: { value: 'WT1' } });
-    });
-
-    const sampleOrGroupOption = screen.getByText(/WT1/);
-    await act(async () => {
-      fireEvent.click(sampleOrGroupOption);
-    });
+    userEvent.click(selectSampleOrGroup);
```

Comment:
Why remove the `act`. We dont need it anymore?

Reply:
userEvent is a library built on top of fireEvent that provides a more user-centric API for triggering events. userEvent automatically wraps all its methods in act(), so you don't need to use act() when using userEvent in most cases. This is one of the reasons why userEvent is recommended for most interaction testing over fireEvent.

In other words, userEvent takes care of using act() internally, ensuring that all state updates and effects are processed when the event is fired, which is why act is not needed anymore in the revised code snippet.

Reply:
it's from chatgpt I think it sums it well